{{cookiecutter.project_slug}}/compose/production/postgres/maintenance/restore

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+
+
+### Restore database from a backup.
+###
+### Parameters:
+###     <1> filename of an existing backup.
+###
+### Usage:
+###     $ docker-compose -f <environment>.yml (exec |run --rm) postgres restore <1>
+
+
+set -o errexit
+set -o pipefail
+set -o nounset
+
+
+working_dir="$(dirname ${0})"
+source "${working_dir}/_sourced/constants.sh"
+source "${working_dir}/_sourced/messages.sh"
+
+
+if [[ -z ${1+x} ]]; then
+    message_error "Backup filename is not specified."
```

Comment:
The error message used to give a hint on how to fix the problem, which is pretty useful. I think it would be good to keep that.

Reply:
Yep, @browniebroke how about that:

```bash
message_error "Restoring as 'postgres' user is not supported. Assign 'POSTGRES_USER' env with another one and try again."
```

Reply:
@browniebroke what would you suggest to replace "Backup filename is not specified." with?

Reply:
My initial point was more about getting a usage reminder like you have in the comment at the top. 

For this specific one, the error message doesn't tell _how_ it should be provided. Is it an option or an argument? I would suggest something along the lines of:
```
message_error "Backup filename is not specified. It must be provided as a positional argument."
```

Reply:
OK, my suggestions are:

```bash
# 1
message_error "Backup filename is not specified yet it is a required parameter. Make sure you provide one and try again."
# 2
message_error "No backup with the specified filename found. Check out the 'backups' maintenance script output to see if there is one and try again."
# 3
message_error "Restoring as 'postgres' user is not supported. Assign 'POSTGRES_USER' env with another one and try again."
```

Reply:
Ok, good for me.